lib/improver/standardise.py

```diff
@@ -0,0 +1,400 @@
+from improver.metadata.check_datatypes import (
+    check_cube_not_float64, check_time_coordinate_metadata)
+from improver.metadata.constants.time_types import (
+    TIME_COORD_NAMES, TIME_REFERENCE_DTYPE, TIME_REFERENCE_UNIT,
+    TIME_DTYPE, TIME_UNIT)
+from improver.threshold import BasicThreshold
+from improver.utilities.cube_checker import spatial_coords_match
+from improver.utilities.spatial import OccurrenceWithinVicinity
+
+
+class StandardiseGridAndMetadata(BasePlugin):
+
+    """Plugin to regrid cube data and standardise metadata"""
+
+    REGRID_REQUIRES_LANDMASK = {"bilinear": False,
+                                "nearest": False,
+                                "nearest-with-mask": True}
+
+    def __init__(self, regrid_mode='bilinear', extrapolation_mode='nanmask',
+                 landmask=None, landmask_vicinity=25000, grid_attributes=None):
+        """
+        Initialise regridding parameters
+
+        Args:
+            regrid_mode (str):
+                Mode of interpolation in regridding.
+            extrapolation_mode (str):
+                Mode to fill regions outside the domain in regridding.
+            landmask (iris.cube.Cube or None):
+                Land-sea mask ("land_binary_mask") on the input cube grid.
+                Required for "nearest-with-mask" regridding option.
+            landmask_vicinity (float):
+                Radius of vicinity to search for a coastline, in metres
+            grid_attributes (list of str or None):
+                List of attribute names to inherit from the target grid cube,
+                eg mosg__model_configuration, that describe the new grid. If
+                None, a list of Met Office-specific attributes is used.
+        """
+        if landmask is None and "nearest-with-mask" in regrid_mode:
+            msg = ("An argument has been specified that requires an input "
+                   "landmask cube but none has been provided")
+            raise ValueError(msg)
+        self.regrid_mode = regrid_mode
+        self.extrapolation_mode = extrapolation_mode
+        self.landmask_source_grid = landmask
+        self.landmask_vicinity = (
+            None if landmask is None else landmask_vicinity)
+        self.landmask_name = "land_binary_mask"
+
+        self.grid_attributes = grid_attributes
+        if self.grid_attributes is None:
+            self.grid_attributes = [
+                'mosg__grid_version', 'mosg__grid_domain', 'mosg__grid_type',
+                'mosg__model_configuration', 'institution']
+
+    def _regrid_landsea(self, cube, target_grid):
+        """
+        Apply land-sea masking to the regridded cube. Raise warnings if
+        landmask metadata is not as expected.
+
+        Returns:
+            iris.cube.Cube: Regridded cube
+        """
+        if self.landmask_name not in self.landmask_source_grid.name():
+            msg = ("Expected land_binary_mask in input_landmask cube "
```

Comment:
Replace land_binary_mask with self.landmask_name in this msg and five lines later.